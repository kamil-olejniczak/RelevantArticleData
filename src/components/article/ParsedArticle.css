.ParsedArticle {
    float: left;
    width: 70%;
    text-align: justify;
    margin: 20px 0 50px;
    padding: 10px 10px;
    position: sticky;
    background-color: white;
    border-radius: 5px;
}

img {
    width: 100%
}

.ParsedArticle figure {
    margin: 0;
}

@media only screen and (max-width: 1670px) {
    .ParsedArticle {
        float: none;
        display: block;
        margin: 20px auto;
        width: 90%;
    }
}