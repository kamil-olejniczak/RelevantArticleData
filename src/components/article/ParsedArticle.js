import React from 'react';
import './ParsedArticle.css';

const ParsedArticle = props => {
    return (
        <div className="ParsedArticle" dangerouslySetInnerHTML={{__html: props.content}}></div>
    );
};

export default ParsedArticle;