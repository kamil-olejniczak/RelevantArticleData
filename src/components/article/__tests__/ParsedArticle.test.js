import React from 'react';
import renderer from 'react-test-renderer';
import ParsedArticle from "../ParsedArticle";
import {shallow} from 'enzyme'

describe('ParsedArticle tests', () => {
    it('renders correctly', () => {
        const component = renderer.create(<ParsedArticle content={fakeData}/>).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should render provided html code', () => {
        const wrapper = shallow(<ParsedArticle content={fakeData}/>);

        expect(wrapper.find('div').prop('dangerouslySetInnerHTML')).toEqual({__html: fakeData});
    });
});

const fakeData = '<div><p>Fake Data</p></div>';
