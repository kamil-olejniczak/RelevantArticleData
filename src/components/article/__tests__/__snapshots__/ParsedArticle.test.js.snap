// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParsedArticle tests renders correctly 1`] = `
<div
  className="ParsedArticle"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div><p>Fake Data</p></div>",
    }
  }
/>
`;
